lib/service/service.go

```diff
@@ -514,6 +510,19 @@ func Run(ctx context.Context, cfg Config, newTeleport NewProcess) error {
 	}
 	// Wait and reload until called exit.
 	for {
+		// Wait for the service to report that it has started.
+		startTimeoutCtx, startCancel := context.WithTimeout(ctx, signalPipeTimeout)
+		defer startCancel()
+		eventC := make(chan Event, 1)
+		srv.WaitForEvent(startTimeoutCtx, TeleportReadyEvent, eventC)
+		select {
+		case <-eventC:
+			cfg.Log.Infof("Service has started successfully.")
+		case <-startTimeoutCtx.Done():
+			warnOnErr(srv.Close(), cfg.Log)
+			return trace.BadParameter("service has failed to start")
+		}
```

Comment:
it looks that `waitAndReload` logic already waits for [TeleportReadyEvent](https://github.com/cove/teleport-exp/blob/b6631a6f886896db42020e174abd7cf1476074dd/lib/service/service.go#L427) so after exiting `waitAndReload` the logic triggers second wait for `TeleportReadyEvent` event but it the TeleportReadyEvent event was ready broadcasted.

Reply:
Ah, good point. In a way this is ok since [an already broadcast event will still be emitted if waited on](https://github.com/gravitational/teleport/blob/16bf416556f337b045b66dc9c3f5a3e16f8cc988/lib/service/supervisor.go#L417-L421), but there might be a better alternative.

Reply:
Doesn't this PR remove that `WaitForEvent`?

Reply:
(I think that a similar situation would already occur if two process reloads were triggered.)

Reply:
> Doesn't this PR remove that `WaitForEvent`?

You're right. Sorry @smallinsky @espadolini, I misunderstood what `waitAndReload` was doing.

I pushed a new change (f73af1864f41125b9522aeaaa1c32ef71fce605e) to ensure that we always for the the `TeleportReadyEvent` after calling `LocalSupervisor.Start()` (both on the initial startup, and after a process reload), reverting the changes to `waitAndReload`.